.github/workflows/frontend.yml

```diff
@@ -186,7 +179,7 @@ jobs:
       # it reduces large coverage fluctuations.
       - name: Handle artifacts
         uses: ./.github/actions/artifacts
-        if: ${{ always() && needs.files-changed.outputs.frontend_all == 'true' }}
+        if: always()
```

Comment:
The `needs.files-changed.*` condition here is moved up to the job, `if: needs.files-changed.outputs.frontend_src == 'true'` on line 46.

No need to repeat.